helper/helper_test.go

```diff
@@ -1,12 +1,187 @@
 package helper
 
 import (
+	"bufio"
+	"fmt"
+	"os"
+
 	"github.com/greenplum-db/gpbackup/utils"
+	"golang.org/x/sys/unix"
 
 	. "github.com/onsi/ginkgo/v2"
 	. "github.com/onsi/gomega"
+
+	"github.com/greenplum-db/gpbackup/toc"
+	"github.com/pkg/errors"
+)
+
+var (
+	testDir     = "/tmp/helper_test/20180101/20180101010101"
+	testTocFile = fmt.Sprintf("%s/test_toc.yaml", testDir)
 )
 
+type restoreReaderTestImpl struct {
+	waitCount int
+}
+
+func (r *restoreReaderTestImpl) waitForPlugin() error {
+	r.waitCount++
+	return nil
+}
+
+func (r *restoreReaderTestImpl) positionReader(pos uint64, oid int) error {
+	return nil
+}
+
+func (r *restoreReaderTestImpl) copyData(num int64) (int64, error) {
+	return 1, nil
+}
+
+func (r *restoreReaderTestImpl) copyAllData() (int64, error) {
+	return 1, nil
+}
+
+func (r *restoreReaderTestImpl) closeFileHandle() {
+}
+
+func (r *restoreReaderTestImpl) getReaderType() ReaderType {
+	return "nil"
+}
+
+type helperTestStep struct {
+	restorePipeWriterArgExpect string
+	restorePipeWriterResult    bool
+	skipFileArgTableOid        int
+	skipFileResult             bool
+	comment                    string
+}
+
+type restoreMockHelperImpl struct {
+	currentStep      int
+	started          bool
+	expectedOidBatch []oidWithBatch
+	expectedSteps    []helperTestStep
+
+	pipesMap    map[string]bool
+	restoreData *restoreReaderTestImpl
+}
+
+func (h *restoreMockHelperImpl) openedPipes() []string {
+	if h.pipesMap == nil {
+		h.pipesMap = make(map[string]bool)
+
+		for k := range pipesMap {
```

Comment:
Why do we need this loop?
the pipe map for test mock is filled in `func (h *restoreMockHelperImpl) createPipe(pipe string)`...

Reply:
If, for some reason, it was not, then, instead of rising error, we create it here and use.

Reply:
what would the `pipesMap` contain in this case when we try to iterate over it?